### DISCUSSION TITLE
How to make the mail button at the bottom of the page work

### DISCUSSION THREAD

**Discussion by Ledmington** [cotes2020/jekyll-theme-chirpy#2009] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Describe the bug

This is not a bug (I hope), it's more of a question on how to fix this thing because I couldn't figure it out.
The mail button at the bottom just displays `javascript:location.href = 'mailto:' + ['filippo.barbari','gmail.com'].join('@')` when hovered (like the one in the demo) but when clicked it doesn't do anything other than opening a new blank tab in my browser.

### Expected Behavior

A new window of the default mailing app should open, like the demo does.

### Environment

- Ruby: `ruby 3.3.4 (2024-07-09 revision be1089c8ec) [x86_64-linux]`
- Jekyll: `jekyll 4.3.4`
- Chirpy: `jekyll-theme-chirpy (7.1.1)`

**Comment by kungfux** [comment]
Are you using custom button definition, because I can see a difference in generated HTML?

Chirpy demo:
```html
<a href="..." aria-label="email">
```

Yours:
```html
<a href="..." aria-label="email" target="_blank" rel="noopener noreferrer">
```

So, the root cause of this behavior should be related to the `target="_blank" rel="noopener noreferrer"` properties defined for a link.

**Reply by Ledmington** [reply]
I think I have not touched the HTML at all. Maybe I need to add `noblank: true` in the `_data/contact.yml` file?

**Reply by kungfux** [reply, marked as answer]
That's right. Please stick to the default configuration layout in `_data/contact.yml`:

```yaml
- type: email
  icon: "fas fa-envelope"
  noblank: true # open link in current tab
```

**Reply by inikishev** [reply]
Can you please share how have your resolved this this issue?

**Reply by kungfux** [reply]
> Can you please share how have your resolved this this issue?

I think the solution was to have `noblank: true` defined for the email as in the example above.

**Reply by Ledmington** [reply]
Yup, sorry, the solution was to add `noblank: true` in `_data/contact.yml`.
